### PR TITLE
Add DataMet backend for Italian Radar data

### PR DESCRIPTION
**Pull request by wolfidan**
Dear xradar people, this is a first try at adding a new backend to read the data from the Italian radars provided by Gianfranco Vulpiani that we plan to use at the open radar course at ERAD2024.

https://zenodo.org/records/4897697

I must say that I had some trouble understanding all the classes that are required to wrap it into xarray, but I tried my best and it seems to work, as I was able to read all 165 provided files. I have added some tests but they are somewhat simplistic at the moment and only test the basic reader not the wrapper to xarray (other tests seem to do the same things).

The DataMet files are not consistent in which sweep corresponds to which elevation: sometimes sweep 1 could be 16° other times 90°. I find it a bit weird but left it as is and didn't try to remap it. I wonder if this could be an issue when reading them into pyart?

Thank you for your time. 
Next week I will be in vacation and might not answer immediately, sorry about that.

- [x] Closes https://github.com/openradar/erad2024/issues/7
- [x] Tests added
- [x] Changes are documented in `history.md`

### CODE REVIEW

**Comment by kmuehlbauer** [comment]
Thanks @wolfidan! The ruff and numpy2 issues should be resolved now by #177. I'll have a closer look here later.

File: xradar/version.py
Comment (kmuehlbauer) [inline xradar/version.py:1]: @wolfidan Please remove that file. It will be autogenerated in the build process.

Comment (kmuehlbauer) [reply xradar/version.py:1]: I've removed it now. Want to see how things go.

**Comment by kmuehlbauer** [comment]
@wolfidan I've taken the chance to fix this up to green CI. Looks good so far. We might just copy/adapt code/tests from the other backends to finish this.

Thanks for this nice addition!

**Comment by kmuehlbauer** [comment]
This works for me with different elevations. Looks that this is a top down strategy.

```python
import open_radar_data
import xarray as xr
fname = open_radar_data.DATASETS.fetch("H-000-VOL-ILMONTE-201907100700.tar.gz")
with xr.open_dataset(fname, engine="datamet", group="sweep_8") as ds:
    display(ds)
    ds = ds.set_coords("sweep_mode")
    ds = ds.wrl.georef.georeference()
    ds.DBTH.wrl.vis.plot()
```

**Comment by wolfidan** [comment]
That's amazing, thanks a lot @kmuehlbauer, I will add more tests the week after the next to increase codecov

**Comment by wolfidan** [comment]
@kmuehlbauer 
All tests and codecov should be fine now. I still had an issue.

        ds = xr.open_dataset(datamet_file, engine="datamet")

doesn't work. I had to use

       ds = xr.open_dataset(datamet_file, engine=xradar.io.backends.DataMetBackendEntrypoint)

in the unit tests

Maybe you can easily fix that. I'm looking  forward to see you do this, so I can do it next time.
Thanks a lot

File: tests/io/test_io.py [hunk 1]
```diff
@@ -793,6 +794,99 @@ def test_write_odim_source(rainbow_file2):
     assert ds["what"].attrs["source"].decode("utf-8") == "NOD:bewid,WMO:06477"
 
 
+def test_open_datamet_dataset(datamet_file):
+    # open first sweep group
+    ds = xr.open_dataset(
+        datamet_file,
+        group="sweep_0",
+        engine=xradar.io.backends.DataMetBackendEntrypoint,
```
Comment (mgrover1) [inline tests/io/test_io.py:802]:
```suggestion
        engine="datamet",
```

File: tests/io/test_io.py [hunk 2]
```diff
@@ -793,6 +794,99 @@ def test_write_odim_source(rainbow_file2):
...
+    assert set(ds.data_vars) & (
+        sweep_dataset_vars | non_standard_sweep_dataset_vars
+    ) == {"DBTH", "DBZH", "KDP", "PHIDP", "RHOHV", "VRADH", "WRADH", "ZDR"}
+    assert ds.sweep_number == 0
+
+    # open last sweep group
+    ds = xr.open_dataset(
+        datamet_file,
+        group="sweep_10",
+        engine=xradar.io.backends.DataMetBackendEntrypoint,
```
Comment (mgrover1) [inline tests/io/test_io.py:814]:
```suggestion
        engine="datamet",
```

File: tests/io/test_io.py [hunk 3]
```diff
@@ -793,6 +794,99 @@ def test_write_odim_source(rainbow_file2):
...
+    assert ds.sweep_number == 10
+
+
+def test_open_datamet_dataset_reindex(datamet_file):
+    # open first sweep group
+    reindex_angle = dict(start_angle=0, stop_angle=360, angle_res=2.0, direction=1)
+    ds = xr.open_dataset(
+        datamet_file,
+        group="sweep_10",
+        engine=xradar.io.backends.DataMetBackendEntrypoint,
```
Comment (mgrover1) [inline tests/io/test_io.py:829]:
```suggestion
        engine="datamet",
```

**Review by mgrover1 — CHANGES_REQUESTED** [review]
Can you try adding back in the engine="datamet" now that the linting issues have been resolved?

File: tests/io/test_io.py (hunk 2 above)
Comment (kmuehlbauer) [inline tests/io/test_io.py:814]:
```suggestion
        engine="datamet",
```

File: tests/io/test_io.py (hunk 3 above)
Comment (kmuehlbauer) [inline tests/io/test_io.py:829]:
```suggestion
        engine="datamet",
```

File: tests/io/test_datamet.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    with _get_data_file(datamet_file, "file") as datametfile:
+        print(datametfile)
+        data = datamet.DataMetFile(datametfile)
+        assert data.filename == datametfile
+        print(data.scan_metadata)
+    data.get_sweep(0)
+    return data
+
+
+def test_file_tyes(data):
```
Comment (kmuehlbauer) [inline tests/io/test_datamet.py:20]:
```suggestion
def test_file_types(data):
```

**Review by kmuehlbauer — CHANGES_REQUESTED** [review]
Not sure why it doesn't work for you. But you would need to install the package first before being able to use the entrypoints.

**Comment by wolfidan** [comment]
Thanks @mgrover1 @kmuehlbauer , I'll change it very soon and see if it passes the tests

**Comment by wolfidan** [comment]
Ok I changed according to your propositions.

**Review by kmuehlbauer — APPROVED** [review]
Great work @wolfidan!

**Review by mgrover1 — APPROVED** [review]
Agreed - thanks so much for this contribution @wolfidan !!

**Pull request merged**